crates/driver/src/infra/time.rs

```diff
@@ -0,0 +1,17 @@
+/// The current time.
+#[derive(Debug, Clone, Copy)]
+pub enum Now {
+    /// Return the time according to this machine's clock.
+    Real,
+    /// Always return the same time. Used for testing.
+    Fake(chrono::DateTime<chrono::Utc>),
+}
+
+impl Now {
+    pub fn now(&self) -> chrono::DateTime<chrono::Utc> {
+        match self {
+            Self::Real => chrono::Utc::now(),
+            Self::Fake(time) => time.to_owned(),
```

Comment:
We might need a way to adjust this value during the test eventually for testing time sensitive stuff.

Reply:
I think in that case you could introduce something like:
```rust
pub fn adjust(&self, new_time: chrono::DateTime<chrono::Utc>) {
    match self {
        Self::Fake(time /* : Mutex<chrono::DateTime<chrono::Utc> */) => *time.lock().unwrap() = new_time,
        Self::Read => panic!("attempt to adjust the system clock"),
    }
}
```

I would leave introducing that to when it is needed.

Reply:
Tokio has a concept of fake time btw https://docs.rs/tokio/latest/tokio/time/index.html .